hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -790,15 +790,14 @@ private List<FileStatusAdapter> listStatusVolume(String volumeStr,
     OFSPath ofsStartPath = new OFSPath(startPath, config);
     // list buckets in the volume
     OzoneVolume volume = objectStore.getVolume(volumeStr);
-    UserGroupInformation ugi =
-        UserGroupInformation.createRemoteUser(volume.getOwner());
-    String owner = ugi.getShortUserName();
-    String group = getGroupName(ugi);
     Iterator<? extends OzoneBucket> iter =
         volume.listBuckets(null, ofsStartPath.getBucketName());
     List<FileStatusAdapter> res = new ArrayList<>();
     while (iter.hasNext() && res.size() < numEntries) {
       OzoneBucket bucket = iter.next();
+      UserGroupInformation ugi = UserGroupInformation.createRemoteUser(bucket.getOwner());
+      String owner = ugi.getShortUserName();
+      String group = getGroupName(ugi);
       res.add(getFileStatusAdapterForBucket(bucket, uri, owner, group));
```

Comment:
Can we just move it inside `getFileStatusAdapterForBucket` so that the function takes care of setting owner and group. Similar to `getFileStatusAdapterForVolume`

Reply:
@aswinshakil Nice suggestion. Thanks!